src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.c

```diff
@@ -1191,7 +1191,7 @@ void BossGanon_SetupTowerCutscene(BossGanon* this, GlobalContext* globalCtx) {
         this->csTimer = 0;
         this->csState = 100;
         this->unk_198 = 1;
-        gSaveContext.magic = gSaveContext.unk_13F4;
+        gSaveContext.magic = gSaveContext.magicCapacityDrawn;
```

Comment:
I'd be okay with either, but I have a slight preference to the way I have it now but it’s hard to articulate why.

But I interpret it as the point of `magicCapacity` being to store the “true” value in a sense while `magicCapacityDrawn` gives flexibility on how the capacity is drawn from 0 -> `magicCapacity` or the previous `magicCapacity` -> new `magicCapacity`. Most checks when comparing against magic capacity seems to be against `magicCapacity` instead of `magicCapacityDrawn`, but either work due to them being mostly the same. But I still has a sense that `magicCapacity` is the true value and calling that `magicCapacityTarget` or `magicCapacityMax` seems a bit funny to me. 

Sure `magicCapacityDrawn` is used more to draw, but it also contains info on the magicCapacity which is what it’s used for, and is still in the name, so I don't find it misleading imo.

With the current names, you also get a meaningful distinction between the two variables instead of something like `magicCapacityMax` and `magicCapacity` which seems too similar.

There also doesn’t seemed to be a logic as to whether devs used `magicCapacity` vs `magicCapacityDrawn` (outside z_parameter) since they basically contain the same info 99.5% of the time and the devs names were awful so I’d understand if they mixed them up a little.

Reply:
To note, I followed through on my suggestion of changing `MAGIC_STATE_GROW_METER` to `MAGIC_STATE_SYNC_METER_WIDTH`, but if these variables were to change, maybe it's be worth changing it to something like `MAGIC_STATE_SYNC_CAPACITIES` or something

Reply:
I agree they are both used as the capacity in different instances, but to me it's misleading to have `magicCapacityDrawn` if it gets used in actual game logic. Personally I would interpret that name as something that is used for drawing on the hud, which means I could change its value without affecting the rest of the game, but that would be a big mistake in this case.

Regarding the names I suggested, it's true that `magicCapacity` and `magicCapacityMax`/`magicCapacityTarget` are pretty similar, but I would argue there is no real way around that since they are used for similar purposes like you mentioned, and we can't fix that on our end unfortunately.

Also, looking at the code again, `magicCapacityDrawn` is actually used more than `magicCapacity`, even when only accounting for game logic purposes. Besides debug prints, `magicCapacity` is only read in one place for case `MAGIC_STATE_FILL` in `Magic_Update`, while `magicCapacityDrawn` is used everywhere else, including when adding magic through `Magic_RequestChange`. So imo it makes more sense to treat it as the "true" capacity in almost all cases (hence `magicCapacity`). And maybe we could have the other one as `magicFillCapacity` since that's only thing it's really used for?

Reply:
Right, I see what you mean.

There's also the process of obtaining magic and double magic:
For obtaining normal magic,
`gSaveContext.magicCapacity = MAGIC_NORMAL_METER;`
Then in `Interface_Update`:
```
if (gSaveContext.isMagicAcquired && (gSaveContext.magicLevel == 0)) {
    gSaveContext.magicLevel = gSaveContext.isDoubleMagicAcquired + 1;
    gSaveContext.magicState = MAGIC_STATE_SYNC_METER_WIDTH;
```
Which is true because the `magicLevel` was 0 before, and this `magicState` is the one that steps `magicCapacityDrawn` -> `magicCapacity` (it happens when the hud is hidden so you don't actually visually see the magic bar grown).

And when obtaining double magic:
```
gSaveContext.magicCapacity = MAGIC_DOUBLE_METER;
gSaveContext.magicLevel = 0;
```
They explicitly set magicLevel to 0 so that the same code in `Interface_Update` runs i.e. `(gSaveContext.magicLevel == 0)` passes, and the state becomes `MAGIC_STATE_SYNC_METER_WIDTH`. to step `magicCapacityDrawn` -> `magicCapacity`

In both cases, `magicCapacity` is acting like a target, and `magicCapacityDrawn` does feel like the true magic capacity.


So I kinda like:
`magicCapacity` -> `magicCapacityTarget`
`magicCapacityDrawn` -> `magicCapacity`
as originally had. Then maybe:
`MAGIC_STATE_SYNC_METER_WIDTH` -> `MAGIC_STATE_SYNC_CAPACITY`

Thoughts?

Reply:
Or maybe something like:
`MAGIC_STATE_SYNC_METER_WIDTH` -> 
`MAGIC_STATE_STEP_CAPACITY`
OR
`MAGIC_STATE_UPDATE_CAPACITY`